python/triton/language/core.py

```diff
@@ -1372,10 +1372,8 @@ def _max_combine(a, b):
 @triton.jit
 def _argmax_combine(value1, index1, value2, index2):
     gt = value1 > value2
-    lt = value1 < value2
-    index_min = minimum(index1, index2)
-    index_ret = where(gt, index1, where(lt, index2, index_min))
-    value_ret = maximum(value1, value2)
+    value_ret = where(gt, value1, value2)
+    index_ret = where(gt, index1, index2)
     return value_ret, index_ret
```

Comment:
Just FYI, the behavior of prefering the lowest index pre-dates my rewrite. If you squint at the old code, you will see that it's equivalent
```cpp
    case RedOp::ARGMAX:
      accIndex = select(
          icmp_sgt(acc, cur), accIndex,
          select(icmp_slt(acc, cur), curIndex, smin(accIndex, curIndex)));
      acc = smax(acc, cur);
      break;
```